secret_store/src/key_server.rs

```diff
@@ -143,67 +142,121 @@ impl Drop for KeyServerCore {
 }
 
 #[cfg(test)]
-mod tests {
+pub mod tests {
 	use std::time;
 	use std::sync::Arc;
 	use ethcrypto;
 	use ethkey::{self, Random, Generator};
 	use acl_storage::tests::DummyAclStorage;
 	use key_storage::tests::DummyKeyStorage;
-	use types::all::{ClusterConfiguration, NodeAddress, EncryptionConfiguration};
+	use types::all::{Error, ClusterConfiguration, NodeAddress, RequestSignature, DocumentAddress, DocumentEncryptedKey, DocumentEncryptedKeyShadow};
 	use super::{KeyServer, KeyServerImpl};
 
-	#[test]
-	fn document_key_generation_and_retrievement_works_over_network() {
-		//::util::log::init_log();
+	pub struct DummyKeyServer;
+
+	impl KeyServer for DummyKeyServer {
+		fn generate_document_key(&self, _signature: &RequestSignature, _document: &DocumentAddress, _threshold: usize) -> Result<DocumentEncryptedKey, Error> {
+			unimplemented!()
+		}
+
+		fn document_key(&self, _signature: &RequestSignature, _document: &DocumentAddress) -> Result<DocumentEncryptedKey, Error> {
+			unimplemented!()
+		}
 
-		let num_nodes = 3;
+		fn document_key_shadow(&self, _signature: &RequestSignature, _document: &DocumentAddress) -> Result<DocumentEncryptedKeyShadow, Error> {
+			unimplemented!()
+		}
+	}
+
+	fn make_key_servers(start_port: u16, num_nodes: usize) -> Vec<KeyServerImpl> {
 		let key_pairs: Vec<_> = (0..num_nodes).map(|_| Random.generate().unwrap()).collect();
 		let configs: Vec<_> = (0..num_nodes).map(|i| ClusterConfiguration {
 				threads: 1,
 				self_private: (***key_pairs[i].secret()).into(),
 				listener_address: NodeAddress {
 					address: "127.0.0.1".into(),
-					port: 6060 + (i as u16),
+					port: start_port + (i as u16),
 				},
 				nodes: key_pairs.iter().enumerate().map(|(j, kp)| (kp.public().clone(),
 					NodeAddress {
 						address: "127.0.0.1".into(),
-						port: 6060 + (j as u16),
+						port: start_port + (j as u16),
 					})).collect(),
 				allow_connecting_to_higher_nodes: false,
-				encryption_config: EncryptionConfiguration {
-					key_check_timeout_ms: 10,
-				},
 			}).collect();
 		let key_servers: Vec<_> = configs.into_iter().map(|cfg|
 			KeyServerImpl::new(&cfg, Arc::new(DummyAclStorage::default()), Arc::new(DummyKeyStorage::default())).unwrap()
 		).collect();
 
-		// wait until connections are established
+		// wait until connections are established. It is fast => do not bother with events here
 		let start = time::Instant::now();
+		let mut tried_reconnections = false;
 		loop {
 			if key_servers.iter().all(|ks| ks.cluster().cluster_state().connected.len() == num_nodes - 1) {
 				break;
 			}
-			if time::Instant::now() - start > time::Duration::from_millis(30000) {
-				panic!("connections are not established in 30000ms");
+
+			let old_tried_reconnections = tried_reconnections;
+			let mut fully_connected = true;
+			for key_server in &key_servers {
```

Comment:
I could do, but this is a test. To make traces work I must also add trace initialization in every test which uses this function. So maybe it is ok to leave it as is, until something go wrong?